contrib/cctz-cmake/CMakeLists.txt: CMake Error: No SOURCES given to target: tzdata

Make sure that `git diff` result is empty and you've just pulled fresh master. Try cleaning up cmake cache. Just in case, official build instructions are published here: https://clickhouse.yandex/docs/en/development/build/

**Operating system**
Tlinux 2.3, inside virtual machine. Tlinux is compatible with CentOS, see [here](https://intl.cloud.tencent.com/document/product/213/32485) for detail.

**Cmake version**
cmake version 3.16.4

**Ninja version**
1.10.0

**Compiler name and version**
gcc (GCC) 9.3.1 20200408 (Red Hat 9.3.1-2)

**Full cmake and/or ninja output**
```
$ cmake -D CMAKE_BUILD_TYPE=Debug ..
-- The C compiler identification is GNU 9.3.1
-- The CXX compiler identification is GNU 9.3.1
-- Check for working C compiler: /opt/rh/devtoolset-9/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-9/root/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rh/devtoolset-9/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-9/root/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using custom linker by name: gold
-- IPO/LTO not enabled.
-- CMAKE_BUILD_TYPE: Debug
-- Using objcopy: /opt/rh/devtoolset-9/root/usr/bin/objcopy.
-- Performing Test HAVE_SSE41
-- Performing Test HAVE_SSE41 - Success
-- Performing Test HAVE_SSE42
-- Performing Test HAVE_SSE42 - Success
-- Performing Test HAVE_SSSE3
-- Performing Test HAVE_SSSE3 - Success
-- Performing Test HAVE_AVX
-- Performing Test HAVE_AVX - Success
-- Performing Test HAVE_AVX2
-- Performing Test HAVE_AVX2 - Success
-- Performing Test HAVE_PCLMULQDQ
-- Performing Test HAVE_PCLMULQDQ - Success
-- Performing Test HAVE_POPCNT
-- Performing Test HAVE_POPCNT - Success
-- Performing Test SUPPORTS_CXXFLAG_error
-- Performing Test SUPPORTS_CXXFLAG_error - Success
-- Performing Test SUPPORTS_CFLAG_error
-- Performing Test SUPPORTS_CFLAG_error - Success
-- Default libraries: -nodefaultlibs -lgcc  -lc -lm -lrt -lpthread -ldl
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- The ASM compiler identification is GNU
-- Found assembler: /opt/rh/devtoolset-9/root/usr/bin/cc
-- Looking for sys/random.h
-- Looking for sys/random.h - not found
-- Some symbols from glibc will be replaced for compatibility
-- Performing Test HAVE_WARNING_UNUSED_BUT_SET_VARIABLE
-- Performing Test HAVE_WARNING_UNUSED_BUT_SET_VARIABLE - Success
-- Performing Test HAVE_WARNING_MISSING_ATTRIBUTES
-- Performing Test HAVE_WARNING_MISSING_ATTRIBUTES - Success
-- Performing Test HAVE_WARNING_MAYBE_UNINITIALIZED
-- Performing Test HAVE_WARNING_MAYBE_UNINITIALIZED - Success
-- Performing Test HAVE_WARNING_REGISTER
-- Performing Test HAVE_WARNING_REGISTER - Success
-- Using libunwind: unwind
-- Using exception handler: unwind
-- Using libcxx: cxx
-- Using libcxxfs: 
-- Using libcxxabi: cxxabi
-- Tests are enabled
-- Building for: Linux-3.10.107-1-tlinux2_kvm_guest-0049 x86_64  ; USE_STATIC_LIBRARIES=ON MAKE_STATIC_LIBRARIES=ON SPLIT_SHARED= UNBUNDLED=OFF CCACHE=/usr/local/bin/ccache 
-- Using double-conversion:  : double-conversion
-- Using ssl=1: /home/pingyu/workspace/ClickHouse/contrib/openssl/include;/home/pingyu/workspace/ClickHouse/contrib/openssl-cmake/linux_x86_64/include : ssl;crypto
-- Using ldap=1: /home/pingyu/workspace/ClickHouse/contrib/openldap-cmake/linux_x86_64/include;/home/pingyu/workspace/ClickHouse/contrib/openldap/include : ldap_r;lber
-- Using icu=1:  : icui18n;icuuc;icudata
-- Using zlib-ng: /home/pingyu/workspace/ClickHouse/contrib/zlib-ng;/home/pingyu/workspace/ClickHouse/build/contrib/zlib-ng : zlib
-- Using zstd: /home/pingyu/workspace/ClickHouse/contrib/zstd/lib : zstd
-- Using sparsehash: /home/pingyu/workspace/ClickHouse/contrib/sparsehash-c11
-- Using re2: /home/pingyu/workspace/ClickHouse/contrib/re2 : re2;  : re2_st
-- Using krb5=1: /home/pingyu/workspace/ClickHouse/contrib/krb5/src/include;/home/pingyu/workspace/ClickHouse/build/contrib/krb5-cmake/include : krb5
-- Using libgsasl=1: /home/pingyu/workspace/ClickHouse/contrib/libgsasl/src;/home/pingyu/workspace/ClickHouse/contrib/libgsasl/linux_x86_64/include : libgsasl
-- Using cyrus-sasl: krb5=1: /home/pingyu/workspace/ClickHouse/contrib/cyrus-sasl/include : sasl2
-- Using librdkafka=1: /home/pingyu/workspace/ClickHouse/contrib/librdkafka/src : rdkafka cppkafka
-- Using AMQP-CPP=1: /home/pingyu/workspace/ClickHouse/contrib/AMQP-CPP/include;/home/pingyu/workspace/ClickHouse/contrib/AMQP-CPP/include;/home/pingyu/workspace/ClickHouse/contrib/AMQP-CPP : AMQP-CPP
-- Using capnp=1: capnpc
-- LLVM include Directory: /home/pingyu/workspace/ClickHouse/contrib/llvm/llvm/include;/home/pingyu/workspace/ClickHouse/build/contrib/llvm/llvm/include
-- LLVM library Directory: /home/pingyu/workspace/ClickHouse/build/contrib/llvm/llvm
-- LLVM C++ compiler flags: 
-- Using h3=1: /home/pingyu/workspace/ClickHouse/contrib/h3/src/h3lib/include : h3
-- Using libxml2: /home/pingyu/workspace/ClickHouse/contrib/libxml2/include;/home/pingyu/workspace/ClickHouse/contrib/libxml2-cmake/linux_x86_64/include : libxml2
-- Using brotli=1: /home/pingyu/workspace/ClickHouse/contrib/brotli/c/include : brotli
-- Using protobuf=1: /home/pingyu/workspace/ClickHouse/contrib/protobuf/src : libprotobuf : $<TARGET_FILE:protoc>
-- Using gRPC=ON: /home/pingyu/workspace/ClickHouse/contrib/grpc/include : libgrpc++
-- Using pdqsort: /home/pingyu/workspace/ClickHouse/contrib/pdqsort
-- Using hdfs3=1: /home/pingyu/workspace/ClickHouse/contrib/libhdfs3/include : hdfs3
-- Using curl: /home/pingyu/workspace/ClickHouse/contrib/curl/include : curl
-- Using aws_s3=1: /home/pingyu/workspace/ClickHouse/contrib/aws/aws-cpp-sdk-s3/include : aws_s3
-- Using snappy: snappy
-- Using Parquet: arrow=arrow_static: ; parquet=parquet_static: ; thrift=thrift_static ; flatbuffers=flatbuffers
-- Using simdjson=ON
-- Using rapidjson=1: /home/pingyu/workspace/ClickHouse/contrib/rapidjson/include
-- Using fastops=1: /home/pingyu/workspace/ClickHouse/contrib/fastops/ : fastops
-- Using unixodbc
-- Using snappy: snappy
-- Using internal=1 orc=1: /home/pingyu/workspace/ClickHouse/contrib/orc/c++/include : orc
-- Using snappy: snappy
-- Using avro=1: /home/pingyu/workspace/ClickHouse/contrib/avro/lang/c++/include : avrocpp
-- Using msgpack=1: /home/pingyu/workspace/ClickHouse/contrib/msgpack-c/include
-- Using cassandra=1: /home/pingyu/workspace/ClickHouse/contrib/cassandra/include/ : cassandra_static
-- Using libuv: /home/pingyu/workspace/ClickHouse/contrib/libuv : uv_a
-- Using sentry=ON: sentry
-- Using stats=1 : /home/pingyu/workspace/ClickHouse/contrib/stats/include
-- Using gcem=1: /home/pingyu/workspace/ClickHouse/contrib/gcem/include
-- Using cityhash:  : cityhash
-- Using farmhash:  : farmhash
-- Using btrie:  : btrie
-- Using gtest=1: /home/pingyu/workspace/ClickHouse/contrib/googletest/googletest : gtest_main;gtest : 
-- Using mysqlclient=1:  : mariadbclient; staticlib=
-- compiler C   = /opt/rh/devtoolset-9/root/usr/bin/cc  -fdiagnostics-color=always  -pipe -msse4.1 -msse4.2 -mpopcnt  -Wall  -Werror -g -O0 -g3 -ggdb3 -fno-inline 
-- compiler CXX = /opt/rh/devtoolset-9/root/usr/bin/c++  -fdiagnostics-color=always -std=c++2a -fsized-deallocation  -pipe -msse4.1 -msse4.2 -mpopcnt   -Wall  -Werror -g -O0 -g3 -ggdb3 -fno-inline  -D_LIBCPP_DEBUG=0
-- LINKER_FLAGS =  -fuse-ld=gold -rdynamic -Wl,--no-undefined -Wl,-no-pie 
-- Packaging with tzdata version: 2020a
find: Arguments to -type should contain only one letter
-- Using cctz
-- Using FastMemcpy
-- Using hyperscan
-- jemalloc malloc_conf: percpu_arena:percpu,oversize_threshold:0,muzzy_decay_ms:10000
-- Using jemalloc
-- Using cpuid
-- Using replxx
-- Using Poco::Crypto
-- Using Poco::Data::ODBC
-- ZSTD VERSION 1.4.0
-- Using CMake version 3.16.4
-- ZLIB_HEADER_VERSION: 1.2.11
-- ZLIBNG_HEADER_VERSION: 1.9.9
-- Arch detected: 'x86_64'
-- Basearch of 'x86_64' has been detected as: 'x86'
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Looking for sys/sdt.h
-- Looking for sys/sdt.h - found
-- Check size of off64_t
-- Check size of off64_t - done
-- Looking for fseeko
-- Looking for fseeko - found
-- Looking for strerror
-- Looking for strerror - found
-- Looking for unistd.h
-- Looking for unistd.h - found
-- Performing Test HAVE_ATTRIBUTE_VISIBILITY_HIDDEN
-- Performing Test HAVE_ATTRIBUTE_VISIBILITY_HIDDEN - Success
-- Performing Test HAVE_ATTRIBUTE_VISIBILITY_INTERNAL
-- Performing Test HAVE_ATTRIBUTE_VISIBILITY_INTERNAL - Success
-- Performing Test HAVE_BUILTIN_CTZL
-- Performing Test HAVE_BUILTIN_CTZL - Success
-- Performing Test HAVE_PTRDIFF_T
-- Performing Test HAVE_PTRDIFF_T - Success
-- Performing Test HAVE_SSE2_INTRIN
-- Performing Test HAVE_SSE2_INTRIN - Success
-- Performing Test HAVE_SSE42CRC_INLINE_ASM
-- Performing Test HAVE_SSE42CRC_INLINE_ASM - Success
-- Performing Test HAVE_SSE42CRC_INTRIN
-- Performing Test HAVE_SSE42CRC_INTRIN - Success
-- Performing Test HAVE_PCLMULQDQ_INTRIN
-- Performing Test HAVE_PCLMULQDQ_INTRIN - Success
-- Architecture-specific source files: arch/x86/x86.c;arch/x86/insert_string_sse.c;arch/x86/deflate_quick.c;arch/x86/fill_window_sse.c;arch/x86/slide_sse.c;arch/x86/crc_folding.c
-- The following features have been enabled:

 * CMAKE_BUILD_TYPE, Build type: Debug (selected)
 * ZLIB_COMPAT, Provide a zlib-compatible API
 * WITH_OPTIM, Build with optimisation
 * WITH_NEW_STRATEGIES, Use new strategies
 * SSE42_CRC, Support CRC hash generation using the SSE4.2 instruction set, using "-msse4"
 * SSE42_DEFLATE_QUICK, Support SSE4.2-accelerated quick compression
 * PCLMUL_CRC, Support CRC hash generation using PCLMULQDQ, using "-mpclmul"

-- The following REQUIRED packages have been found:

 * Threads

-- The following features have been disabled:

 * WITH_GZFILEOP, Compile with support for gzFile-related functions
 * WITH_SANITIZERS, Build with address sanitizer and all supported sanitizers other than memory sanitizer
 * WITH_MSAN, Build with memory sanitizer
 * WITH_FUZZERS, Build test/fuzz
 * MAINTAINER, Build with maintainer warnings and tests enabled

-- Found CURL: curl (found version "7.29.0")  
-- Found GSSAPI: -lgssapi_krb5 -lkrb5 -lk5crypto -lcom_err
-- Performing Test HAS_-Wunused_FLAG
-- Performing Test HAS_-Wunused_FLAG - Success
-- Performing Test HAS_-Wlogical-op_FLAG
-- Performing Test HAS_-Wlogical-op_FLAG - Success
-- Performing Test HAS_-Wno-uninitialized_FLAG
-- Performing Test HAS_-Wno-uninitialized_FLAG - Success
-- Performing Test HAS_-Wall_FLAG
-- Performing Test HAS_-Wall_FLAG - Success
-- Performing Test HAS_-Wextra_FLAG
-- Performing Test HAS_-Wextra_FLAG - Success
-- Performing Test HAS_-Wformat-security_FLAG
-- Performing Test HAS_-Wformat-security_FLAG - Success
-- Performing Test HAS_-Wno-init-self_FLAG
-- Performing Test HAS_-Wno-init-self_FLAG - Success
-- Performing Test HAS_-Wwrite-strings_FLAG
-- Performing Test HAS_-Wwrite-strings_FLAG - Success
-- Performing Test HAS_-Wshift-count-overflow_FLAG
-- Performing Test HAS_-Wshift-count-overflow_FLAG - Success
-- Performing Test HAS_-Wdeclaration-after-statement_FLAG
-- Performing Test HAS_-Wdeclaration-after-statement_FLAG - Success
-- Performing Test HAS_-Wno-undef_FLAG
-- Performing Test HAS_-Wno-undef_FLAG - Success
-- Performing Test HAS_-Wno-unknown-pragmas_FLAG
-- Performing Test HAS_-Wno-unknown-pragmas_FLAG - Success
-- Found Git: /usr/local/bin/git (found version "2.20.1") 
-- Check if the system is big endian
-- Searching 16 bit integer
-- Check size of unsigned short
-- Check size of unsigned short - done
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for include file alloca.h
-- Looking for include file alloca.h - found
-- Looking for include file arpa/inet.h
-- Looking for include file arpa/inet.h - found
-- Looking for include file dlfcn.h
-- Looking for include file dlfcn.h - found
-- Looking for include file fcntl.h
-- Looking for include file fcntl.h - found
-- Looking for include file float.h
-- Looking for include file float.h - found
-- Looking for include file limits.h
-- Looking for include file limits.h - found
-- Looking for include file pwd.h
-- Looking for include file pwd.h - found
-- Looking for include file sched.h
-- Looking for include file sched.h - found
-- Looking for include file select.h
-- Looking for include file select.h - not found
-- Looking for include file signal.h
-- Looking for include file signal.h - found
-- Looking for include file stdlib.h
-- Looking for include file stdlib.h - found
-- Looking for include file string.h
-- Looking for include file string.h - found
-- Looking for include file strings.h
-- Looking for include file strings.h - found
-- Looking for include file sys/ioctl.h
-- Looking for include file sys/ioctl.h - found
-- Looking for include file sys/select.h
-- Looking for include file sys/select.h - found
-- Looking for include file sys/socket.h
-- Looking for include file sys/socket.h - found
-- Looking for include file sys/un.h
-- Looking for include file sys/un.h - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for include file utime.h
-- Looking for include file utime.h - found
-- Looking for include file ucontext.h
-- Looking for include file ucontext.h - found
-- Looking for alloca
-- Looking for alloca - not found
-- Looking for dlerror
-- Looking for dlerror - not found
-- Looking for dlopen
-- Looking for dlopen - not found
-- Looking for fcntl
-- Looking for fcntl - found
-- Looking for memcpy
-- Looking for memcpy - found
-- Looking for nl_langinfo
-- Looking for nl_langinfo - found
-- Looking for setlocale
-- Looking for setlocale - found
-- Looking for poll
-- Looking for poll - found
-- Check size of char *
-- Check size of char * - done
-- Check size of int
-- Check size of int - done
-- Check size of long
-- Check size of long - done
-- Check size of long long
-- Check size of long long - done
-- Check size of size_t
-- Check size of size_t - done
-- Check size of uchar
-- Check size of uchar - failed
-- Check size of uint
-- Check size of uint - done
-- Check size of ulong
-- Check size of ulong - done
-- Check size of int8
-- Check size of int8 - failed
-- Check size of uint8
-- Check size of uint8 - failed
-- Check size of int16
-- Check size of int16 - failed
-- Check size of uint16
-- Check size of uint16 - failed
-- Check size of int32
-- Check size of int32 - failed
-- Check size of uint32
-- Check size of uint32 - failed
-- Check size of int64
-- Check size of int64 - failed
-- Check size of uint64
-- Check size of uint64 - failed
-- Check size of socklen_t
-- Check size of socklen_t - failed
-- Performing Test SOCKET_SIZE_FOUND_socklen_t
-- Performing Test SOCKET_SIZE_FOUND_socklen_t - Success
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- TLS library/version: OpenSSL 
-- SYSTEM_LIBS zlib;ssl;crypto
-- Dynamic column API support: ON
-- SYSTEM processor: x86_64
-- MariaDB Connector/c configuration:
-- Static PLUGINS mysql_native_password;caching_sha2_password;sha256_password;mysql_old_password;pvio_socket
-- Dynamic PLUGINS 
-- CPack generation: TGZ
-- SSL support: OPENSSL Libs: ssl;crypto
-- Zlib support: ON
-- Installation layout: DEFAULT
-- Include files will be installed in include/mariadb
-- Libraries will be installed in lib/mariadb
-- Binaries will be installed in bin
-- Documentation included from 
-- Required: 
-- librdkafka with SASL support
-- Check if the system is big endian
-- Searching 16 bit integer
-- Using unsigned short
-- Check if the system is big endian - little endian
-- Looking for byteswap.h
-- Looking for byteswap.h - found
-- Looking for sys/endian.h
-- Looking for sys/endian.h - not found
-- Looking for sys/mman.h
-- Looking for sys/mman.h - found
-- Looking for sys/resource.h
-- Looking for sys/resource.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for sys/uio.h
-- Looking for sys/uio.h - found
-- Looking for windows.h
-- Looking for windows.h - not found
-- Looking for zlibVersion in z
-- Looking for zlibVersion in z - found
-- Looking for lzo1x_1_15_compress in lzo2
-- Looking for lzo1x_1_15_compress in lzo2 - not found
-- Performing Test HAVE_VISUAL_STUDIO_ARCH_AVX
-- Performing Test HAVE_VISUAL_STUDIO_ARCH_AVX - Failed
-- Performing Test HAVE_VISUAL_STUDIO_ARCH_AVX2
-- Performing Test HAVE_VISUAL_STUDIO_ARCH_AVX2 - Failed
-- Performing Test HAVE_CLANG_MAVX
-- Performing Test HAVE_CLANG_MAVX - Success
-- Performing Test HAVE_CLANG_MBMI2
-- Performing Test HAVE_CLANG_MBMI2 - Success
-- Performing Test HAVE_BUILTIN_EXPECT
-- Performing Test HAVE_BUILTIN_EXPECT - Success
-- Performing Test HAVE_BUILTIN_CTZ
-- Performing Test HAVE_BUILTIN_CTZ - Success
-- Performing Test SNAPPY_HAVE_SSSE3
-- Performing Test SNAPPY_HAVE_SSSE3 - Success
-- Performing Test SNAPPY_HAVE_BMI2
-- Performing Test SNAPPY_HAVE_BMI2 - Failed
-- Looking for mmap
-- Looking for mmap - found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for strtof_l
-- Looking for strtof_l - found
-- Looking for strtoull_l
-- Looking for strtoull_l - found
-- FLATBUFFERS_LIBRARY: flatbuffers
-- Performing Test ORC_CXX_HAS_INITIALIZER_LIST
-- Performing Test ORC_CXX_HAS_INITIALIZER_LIST - Success
-- Performing Test ORC_CXX_HAS_NOEXCEPT
-- Performing Test ORC_CXX_HAS_NOEXCEPT - Success
-- Performing Test ORC_CXX_HAS_NULLPTR
-- Performing Test ORC_CXX_HAS_NULLPTR - Success
-- Performing Test ORC_CXX_HAS_OVERRIDE
-- Performing Test ORC_CXX_HAS_OVERRIDE - Success
-- Performing Test ORC_CXX_HAS_UNIQUE_PTR
-- Performing Test ORC_CXX_HAS_UNIQUE_PTR - Success
-- Performing Test ORC_CXX_HAS_CSTDINT
-- Performing Test ORC_CXX_HAS_CSTDINT - Success
-- Performing Test ORC_CXX_HAS_THREAD_LOCAL
-- Performing Test ORC_CXX_HAS_THREAD_LOCAL - Success
-- Performing Test HAS_PREAD
-- Performing Test HAS_PREAD - Success
-- Performing Test HAS_STRPTIME
-- Performing Test HAS_STRPTIME - Success
-- Performing Test HAS_STOLL
-- Performing Test HAS_STOLL - Success
-- Performing Test INT64_IS_LL
-- Performing Test INT64_IS_LL - Success
-- Performing Test HAS_DIAGNOSTIC_PUSH
-- Performing Test HAS_DIAGNOSTIC_PUSH - Success
-- Performing Test HAS_STD_ISNAN
-- Performing Test HAS_STD_ISNAN - Success
-- Performing Test HAS_STD_MUTEX
-- Performing Test HAS_STD_MUTEX - Success
-- Performing Test NEEDS_REDUNDANT_MOVE
-- Performing Test NEEDS_REDUNDANT_MOVE - Success
-- Performing Test HAS_PRE_1970
-- Performing Test HAS_PRE_1970 - Success
-- Performing Test HAS_POST_2038
-- Performing Test HAS_POST_2038 - Success
-- Performing Test NEEDS_Z_PREFIX
-- Performing Test NEEDS_Z_PREFIX - Failed
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Looking for inttypes.h
-- Looking for inttypes.h - found
-- Looking for netdb.h
-- Looking for netdb.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for signal.h
-- Looking for signal.h - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for sys/param.h
-- Looking for sys/param.h - found
-- Looking for sys/stat.h
-- Looking for sys/stat.h - found
-- Looking for poll.h
-- Looking for poll.h - found
-- Looking for sys/poll.h
-- Looking for sys/poll.h - found
-- Looking for gethostbyname
-- Looking for gethostbyname - found
-- Looking for gethostbyname_r
-- Looking for gethostbyname_r - found
-- Looking for strerror_r
-- Looking for strerror_r - found
-- Looking for sched_get_priority_max
-- Looking for sched_get_priority_max - found
-- Looking for sched_get_priority_min
-- Looking for sched_get_priority_min - found
-- Performing Test STRERROR_R_CHAR_P
-- Performing Test STRERROR_R_CHAR_P - Success
-- Found PythonInterp: /usr/bin/python (found version "2.7.5") 
-- Looking for errno.h
-- Looking for errno.h - found
-- Looking for link.h
-- Looking for link.h - found
-- Looking for malloc/malloc.h
-- Looking for malloc/malloc.h - not found
-- Looking for termios.h
-- Looking for termios.h - found
-- Looking for valgrind/valgrind.h
-- Looking for valgrind/valgrind.h - not found
-- Looking for zlib.h
-- Looking for zlib.h - found
-- Looking for fenv.h
-- Looking for fenv.h - found
-- Looking for FE_ALL_EXCEPT
-- Looking for FE_ALL_EXCEPT - found
-- Looking for FE_INEXACT
-- Looking for FE_INEXACT - found
-- Looking for mach/mach.h
-- Looking for mach/mach.h - not found
-- Looking for histedit.h
-- Looking for histedit.h - not found
-- Looking for CrashReporterClient.h
-- Looking for CrashReporterClient.h - not found
-- Looking for linux/magic.h
-- Looking for linux/magic.h - found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Looking for pthread_getspecific in pthread
-- Looking for pthread_getspecific in pthread - found
-- Looking for pthread_rwlock_init in pthread
-- Looking for pthread_rwlock_init in pthread - found
-- Looking for pthread_mutex_lock in pthread
-- Looking for pthread_mutex_lock in pthread - found
-- Looking for dlopen in dl
-- Looking for dlopen in dl - found
-- Looking for clock_gettime in rt
-- Looking for clock_gettime in rt - found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Looking for compress2 in z
-- Looking for compress2 in z - found
-- Looking for xar_open in xar
-- Looking for xar_open in xar - not found
-- Looking for arc4random
-- Looking for arc4random - not found
-- Looking for backtrace
-- Looking for backtrace - found
-- backtrace facility detected in default set of libraries
-- Found Backtrace: /usr/include  
-- Performing Test C_SUPPORTS_WERROR_UNGUARDED_AVAILABILITY_NEW
-- Performing Test C_SUPPORTS_WERROR_UNGUARDED_AVAILABILITY_NEW - Failed
-- Looking for _Unwind_Backtrace
-- Looking for _Unwind_Backtrace - found
-- Looking for getpagesize
-- Looking for getpagesize - found
-- Looking for sysconf
-- Looking for sysconf - found
-- Looking for getrusage
-- Looking for getrusage - found
-- Looking for setrlimit
-- Looking for setrlimit - found
-- Looking for isatty
-- Looking for isatty - found
-- Looking for futimens
-- Looking for futimens - found
-- Looking for futimes
-- Looking for futimes - found
-- Looking for posix_fallocate
-- Looking for posix_fallocate - found
-- Looking for sigaltstack
-- Looking for sigaltstack - found
-- Looking for lseek64
-- Looking for lseek64 - found
-- Looking for mallctl
-- Looking for mallctl - not found
-- Looking for mallinfo
-- Looking for mallinfo - found
-- Looking for malloc_zone_statistics
-- Looking for malloc_zone_statistics - not found
-- Looking for getrlimit
-- Looking for getrlimit - found
-- Looking for posix_spawn
-- Looking for posix_spawn - found
-- Looking for pread
-- Looking for pread - found
-- Looking for sbrk
-- Looking for sbrk - found
-- Looking for strerror_s
-- Looking for strerror_s - not found
-- Looking for setenv
-- Looking for setenv - found
-- Looking for dladdr
-- Looking for dladdr - not found
-- Performing Test HAVE_STRUCT_STAT_ST_MTIMESPEC_TV_NSEC
-- Performing Test HAVE_STRUCT_STAT_ST_MTIMESPEC_TV_NSEC - Failed
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC
-- Performing Test HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC - Success
-- Looking for __GLIBC__
-- Looking for __GLIBC__ - found
-- Looking for sched_getaffinity
-- Looking for sched_getaffinity - found
-- Looking for CPU_COUNT
-- Looking for CPU_COUNT - found
-- Looking for pthread_getname_np
-- Looking for pthread_getname_np - found
-- Looking for pthread_setname_np
-- Looking for pthread_setname_np - found
-- Performing Test HAVE_STD_IS_TRIVIALLY_COPYABLE
-- Performing Test HAVE_STD_IS_TRIVIALLY_COPYABLE - Success
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS_WITHOUT_LIB - Success
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB
-- Performing Test HAVE_CXX_ATOMICS64_WITHOUT_LIB - Success
-- Performing Test LLVM_HAS_ATOMICS
-- Performing Test LLVM_HAS_ATOMICS - Success
-- Performing Test SUPPORTS_NO_PIE_FLAG
-- Performing Test SUPPORTS_NO_PIE_FLAG - Success
-- Performing Test SUPPORTS_VARIADIC_MACROS_FLAG
-- Performing Test SUPPORTS_VARIADIC_MACROS_FLAG - Success
-- Performing Test SUPPORTS_GNU_ZERO_VARIADIC_MACRO_ARGUMENTS_FLAG
-- Performing Test SUPPORTS_GNU_ZERO_VARIADIC_MACRO_ARGUMENTS_FLAG - Failed
-- Performing Test HAS_MAYBE_UNINITIALIZED
-- Performing Test HAS_MAYBE_UNINITIALIZED - Success
-- Native target architecture is X86
-- Threads enabled.
-- Doxygen disabled.
stat /home/pingyu/workspace/ClickHouse/contrib/llvm/llvm/bindings/go/conftest.go: no such file or directory
-- Go bindings disabled, need at least Go 1.2.
-- Ninja version: 1.10.0
-- Could NOT find OCaml (missing: OCAMLFIND OCAML_VERSION OCAML_STDLIB_PATH) 
-- Could NOT find OCaml (missing: OCAMLFIND OCAML_VERSION OCAML_STDLIB_PATH) 
-- OCaml bindings disabled.
-- Found Python module pygments
-- Could NOT find Python module pygments.lexers.c_cpp
-- Found Python module yaml
-- LLVM host triple: x86_64-unknown-linux-gnu
-- LLVM default target triple: x86_64-unknown-linux-gnu
-- Performing Test SUPPORTS_FVISIBILITY_INLINES_HIDDEN_FLAG
-- Performing Test SUPPORTS_FVISIBILITY_INLINES_HIDDEN_FLAG - Success
-- Performing Test C_SUPPORTS_WERROR_DATE_TIME
-- Performing Test C_SUPPORTS_WERROR_DATE_TIME - Success
-- Performing Test CXX_SUPPORTS_WERROR_DATE_TIME
-- Performing Test CXX_SUPPORTS_WERROR_DATE_TIME - Success
-- Performing Test CXX_SUPPORTS_WERROR_UNGUARDED_AVAILABILITY_NEW
-- Performing Test CXX_SUPPORTS_WERROR_UNGUARDED_AVAILABILITY_NEW - Failed
-- Performing Test CXX_SUPPORTS_CXX_STD
-- Performing Test CXX_SUPPORTS_CXX_STD - Success
-- Performing Test CXX_SUPPORTS_MISSING_FIELD_INITIALIZERS_FLAG
-- Performing Test CXX_SUPPORTS_MISSING_FIELD_INITIALIZERS_FLAG - Success
-- Performing Test C_SUPPORTS_IMPLICIT_FALLTHROUGH_FLAG
-- Performing Test C_SUPPORTS_IMPLICIT_FALLTHROUGH_FLAG - Success
-- Performing Test CXX_SUPPORTS_IMPLICIT_FALLTHROUGH_FLAG
-- Performing Test CXX_SUPPORTS_IMPLICIT_FALLTHROUGH_FLAG - Success
-- Performing Test C_SUPPORTS_COVERED_SWITCH_DEFAULT_FLAG
-- Performing Test C_SUPPORTS_COVERED_SWITCH_DEFAULT_FLAG - Failed
-- Performing Test CXX_SUPPORTS_COVERED_SWITCH_DEFAULT_FLAG
-- Performing Test CXX_SUPPORTS_COVERED_SWITCH_DEFAULT_FLAG - Failed
-- Performing Test CXX_SUPPORTS_CLASS_MEMACCESS_FLAG
-- Performing Test CXX_SUPPORTS_CLASS_MEMACCESS_FLAG - Success
-- Performing Test CXX_SUPPORTS_NOEXCEPT_TYPE_FLAG
-- Performing Test CXX_SUPPORTS_NOEXCEPT_TYPE_FLAG - Success
-- Performing Test C_SUPPORTS_DELETE_NON_VIRTUAL_DTOR_FLAG
-- Performing Test C_SUPPORTS_DELETE_NON_VIRTUAL_DTOR_FLAG - Failed
-- Performing Test CXX_SUPPORTS_DELETE_NON_VIRTUAL_DTOR_FLAG
-- Performing Test CXX_SUPPORTS_DELETE_NON_VIRTUAL_DTOR_FLAG - Success
-- Performing Test C_WCOMMENT_ALLOWS_LINE_WRAP
-- Performing Test C_WCOMMENT_ALLOWS_LINE_WRAP - Success
-- Performing Test C_SUPPORTS_STRING_CONVERSION_FLAG
-- Performing Test C_SUPPORTS_STRING_CONVERSION_FLAG - Failed
-- Performing Test CXX_SUPPORTS_STRING_CONVERSION_FLAG
-- Performing Test CXX_SUPPORTS_STRING_CONVERSION_FLAG - Failed
-- Performing Test LINKER_SUPPORTS_COLOR_DIAGNOSTICS
-- Performing Test LINKER_SUPPORTS_COLOR_DIAGNOSTICS - Failed
-- Looking for os_signpost_interval_begin
-- Looking for os_signpost_interval_begin - not found
-- Constructing LLVMBuild project information
-- Linker detection: GNU Gold
-- LLVMHello ignored -- Loadable modules not supported on this platform.
-- Targeting X86
-- Targeting AArch64
-- checking compiler: GCC (9.3.)
-- Looking for nanosleep
-- Looking for nanosleep - found
-- Checking whether strerror_r returns an int
-- Checking whether strerror_r returns an int -- no
-- Looking for res_servicename in resolv
-- Looking for res_servicename in resolv - not found
-- Looking for gethostbyname in nsl
-- Looking for gethostbyname in nsl - found
-- Looking for gethostbyname in socket
-- Looking for gethostbyname in socket - not found
-- Looking for socket in socket
-- Looking for socket in socket - not found
-- Looking for include file arpa/nameser_compat.h
-- Looking for include file arpa/nameser_compat.h - found
-- Looking for include file arpa/nameser.h
-- Looking for include file arpa/nameser.h - found
-- Looking for include file assert.h
-- Looking for include file assert.h - found
-- Looking for include file malloc.h
-- Looking for include file malloc.h - found
-- Looking for include file memory.h
-- Looking for include file memory.h - found
-- Looking for include file netinet/tcp.h
-- Looking for include file netinet/tcp.h - found
-- Looking for include file net/if.h
-- Looking for include file net/if.h - found
-- Looking for include file socket.h
-- Looking for include file socket.h - not found
-- Looking for include file stdbool.h
-- Looking for include file stdbool.h - found
-- Looking for include file stropts.h
-- Looking for include file stropts.h - not found
-- Looking for include file time.h
-- Looking for include file time.h - found
-- Looking for include files winsock2.h, windows.h
-- Looking for include files winsock2.h, windows.h - not found
-- Looking for 3 include files winsock2.h, ..., windows.h
-- Looking for 3 include files winsock2.h, ..., windows.h - not found
-- Looking for include files winsock.h, windows.h
-- Looking for include files winsock.h, windows.h - not found
-- Performing Test HAVE_SOCKLEN_T
-- Performing Test HAVE_SOCKLEN_T - Success
-- Performing Test HAVE_TYPE_SOCKET
-- Performing Test HAVE_TYPE_SOCKET - Failed
-- Performing Test HAVE_BOOL_T
-- Performing Test HAVE_BOOL_T - Success
-- Performing Test HAVE_SSIZE_T
-- Performing Test HAVE_SSIZE_T - Success
-- Performing Test HAVE_LONGLONG
-- Performing Test HAVE_LONGLONG - Success
-- Performing Test HAVE_SIG_ATOMIC_T
-- Performing Test HAVE_SIG_ATOMIC_T - Success
-- Performing Test HAVE_STRUCT_ADDRINFO
-- Performing Test HAVE_STRUCT_ADDRINFO - Success
-- Performing Test HAVE_STRUCT_IN6_ADDR
-- Performing Test HAVE_STRUCT_IN6_ADDR - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6
-- Performing Test HAVE_STRUCT_SOCKADDR_IN6 - Success
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE
-- Performing Test HAVE_STRUCT_SOCKADDR_STORAGE - Success
-- Performing Test HAVE_STRUCT_TIMEVAL
-- Performing Test HAVE_STRUCT_TIMEVAL - Success
-- Looking for AF_INET6
-- Looking for AF_INET6 - found
-- Looking for O_NONBLOCK
-- Looking for O_NONBLOCK - found
-- Looking for FIONBIO
-- Looking for FIONBIO - found
-- Looking for SIOCGIFADDR
-- Looking for SIOCGIFADDR - found
-- Looking for MSG_NOSIGNAL
-- Looking for MSG_NOSIGNAL - found
-- Looking for PF_INET6
-- Looking for PF_INET6 - found
-- Looking for SO_NONBLOCK
-- Looking for SO_NONBLOCK - not found
-- Looking for CLOCK_MONOTONIC
-- Looking for CLOCK_MONOTONIC - found
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID
-- Performing Test HAVE_SOCKADDR_IN6_SIN6_SCOPE_ID - Success
-- Performing Test HAVE_LL
-- Performing Test HAVE_LL - Success
-- Looking for bitncmp
-- Looking for bitncmp - not found
-- Looking for closesocket
-- Looking for closesocket - not found
-- Looking for CloseSocket
-- Looking for CloseSocket - not found
-- Looking for connect
-- Looking for connect - found
-- Looking for freeaddrinfo
-- Looking for freeaddrinfo - found
-- Looking for getaddrinfo
-- Looking for getaddrinfo - found
-- Looking for getenv
-- Looking for getenv - found
-- Looking for gethostbyaddr
-- Looking for gethostbyaddr - found
-- Looking for gethostname
-- Looking for gethostname - found
-- Looking for getnameinfo
-- Looking for getnameinfo - found
-- Looking for getservbyport_r
-- Looking for getservbyport_r - found
-- Looking for gettimeofday
-- Looking for gettimeofday - found
-- Looking for if_indextoname
-- Looking for if_indextoname - found
-- Looking for inet_net_pton
-- Looking for inet_net_pton - not found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Looking for inet_pton
-- Looking for inet_pton - found
-- Looking for ioctl
-- Looking for ioctl - found
-- Looking for ioctlsocket
-- Looking for ioctlsocket - not found
-- Looking for IoctlSocket
-- Looking for IoctlSocket - not found
-- Looking for recv
-- Looking for recv - found
-- Looking for recvfrom
-- Looking for recvfrom - found
-- Looking for send
-- Looking for send - found
-- Looking for setsockopt
-- Looking for setsockopt - found
-- Looking for socket
-- Looking for socket - found
-- Looking for strcasecmp
-- Looking for strcasecmp - found
-- Looking for strcmpi
-- Looking for strcmpi - not found
-- Looking for strdup
-- Looking for strdup - found
-- Looking for stricmp
-- Looking for stricmp - not found
-- Looking for strncasecmp
-- Looking for strncasecmp - found
-- Looking for strncmpi
-- Looking for strncmpi - not found
-- Looking for strnicmp
-- Looking for strnicmp - not found
-- Looking for writev
-- Looking for writev - found
-- Looking for __system_property_get
-- Looking for __system_property_get - not found
-- PROJECT version: 2.11.0
-- libuv version: v1.34.1
-- Found OpenSSL: ssl  
-- OpenSSL version: v
-- Using std::atomic implementation for atomic operations
-- Checking to see if CXX compiler accepts flag -std=c++11
-- Checking to see if CXX compiler accepts flag -std=c++11 - yes
-- Using hash header <functional> and namespace "std"
-- Check size of long long
-- Check size of long long - done
-- Check size of uint16_t
-- Check size of uint16_t - done
-- Check size of u_int16_t
-- Check size of u_int16_t - done
-- Check size of __uint16_t
-- Check size of __uint16_t - done
-- Looking for GRND_NONBLOCK
-- Looking for GRND_NONBLOCK - not found
-- Looking for SO_NOSIGPIPE
-- Looking for SO_NOSIGPIPE - not found
-- Looking for sigtimedwait
-- Looking for sigtimedwait - found
-- Performing Test HAVE_BUILTIN_BSWAP32
-- Performing Test HAVE_BUILTIN_BSWAP32 - Success
-- Performing Test HAVE_BUILTIN_BSWAP64
-- Performing Test HAVE_BUILTIN_BSWAP64 - Success
-- SENTRY_TRANSPORT=curl
-- SENTRY_BACKEND=none
-- Check size of long
-- Check size of long - done
-- GCE-Math version 1.13.0
-- /home/pingyu/workspace/ClickHouse/src: Have 10593 megabytes of memory. Limiting concurrent linkers jobs to 3 and compiler jobs to 4
-- Will build ClickHouse 20.8.1.1 revision 54438 
-- Target check already exists
-- /home/pingyu/workspace/ClickHouse/utils: Have 10587 megabytes of memory. Limiting concurrent linkers jobs to 3 and compiler jobs to OFF
-- Configuring done
CMake Error at contrib/cctz-cmake/CMakeLists.txt:111 (add_library):
  No SOURCES given to target: tzdata


CMake Generate step failed.  Build files cannot be regenerated correctly.
```


Notice the following lines:
```
-- Packaging with tzdata version: 2020a
find: Arguments to -type should contain only one letter

...

CMake Error at contrib/cctz-cmake/CMakeLists.txt:111 (add_library):
  No SOURCES given to target: tzdata
```
The error is produced [here](https://github.com/ClickHouse/ClickHouse/blob/master/contrib/cctz-cmake/CMakeLists.txt#L85):
> bash -c "cd ${TZDIR} && find * -type f,l -and ! -name '*.tab' -and ! -name 'localtime' | sort | paste -sd ';'"


On my machine, `find` with version `find (GNU findutils) 4.5.11` do not support multiple types specified by `-type`, which is a [GNU extension](https://man7.org/linux/man-pages/man1/find.1.html).
```
[zoneinfo] $ find * -type f,l -and ! -name '*.tab' -and ! -name 'localtime' | sort | paste -sd ';'
find: Arguments to -type should contain only one letter
```


Modify `find * -type f,l` to `find -L * -type f` will fix the issue:
```
[zoneinfo] $ find -L * -type f -and ! -name '*.tab' -and ! -name 'localtime' | sort | paste -sd ';'
Africa/Abidjan;Africa/Accra;Africa/Addis_Ababa;Africa/Algiers;Africa/...
```
And I think `find -L -type f` would be better than `find -type f,l`, which still produce correct result for situation such as symbolic link to directory.

--- a/contrib/cctz-cmake/CMakeLists.txt
+++ b/contrib/cctz-cmake/CMakeLists.txt
@@ -82,7 +82,7 @@ if (NOT EXTERNAL_CCTZ_LIBRARY_FOUND OR NOT EXTERNAL_CCTZ_LIBRARY_WORKS)
 
         # each file/symlink in that dir (except of tab and localtime) store the info about timezone
         execute_process(COMMAND
-            bash -c "cd ${TZDIR} && find * -type f,l -and ! -name '*.tab' -and ! -name 'localtime' | sort | paste -sd ';'"
+            bash -c "cd ${TZDIR} && find * -type f -and ! -name '*.tab' -and ! -name 'localtime' | sort | paste -sd ';'"
             OUTPUT_STRIP_TRAILING_WHITESPACE
             OUTPUT_VARIABLE TIMEZONES)
 

--- a/contrib/cctz-cmake/CMakeLists.txt
+++ b/contrib/cctz-cmake/CMakeLists.txt
@@ -80,7 +80,7 @@ if (NOT EXTERNAL_CCTZ_LIBRARY_FOUND OR NOT EXTERNAL_CCTZ_LIBRARY_WORKS)
 
         set(TZ_OBJS)
 
-        # each file/symlink in that dir (except of tab and localtime) store the info about timezone
+        # each file in that dir (except of tab and localtime) store the info about timezone
         execute_process(COMMAND
             bash -c "cd ${TZDIR} && find * -type f -and ! -name '*.tab' -and ! -name 'localtime' | sort | paste -sd ';'"
             OUTPUT_STRIP_TRAILING_WHITESPACE

